src/sygnet/sygnet_interface.py

```diff
@@ -212,7 +231,7 @@ def fit(
                 batch_size = batch_size,
                 learning_rate = learning_rate,
                 use_tensorboard = use_tensorboard
-            )
```

Comment:
Do we need the pickle dump as a return object? Perhaps this is as a result of my awkward coding: I made the train_* functions return None in case anyone tries to assign the result i.e. `new_var = model.fit(...)`.

Anyway, since this code is replicated on lines 247 and 260, I wonder if we can just have a separate conditional at the end of the .fit() method, i.e.:

```
if save_model = True:
   pickle.dump(...
```

Reply:
Tom, I have done the .idea and the toml file.
As for 
if save_model = True:
   pickle.dump(...
The way it is done now, lets us capture all of the model object (using self)
Calling such conditional you propose can only be done inside a train fucntion. We will need to pass save_model = save_model few levels down to the train(), but that's fine. But if we call pickle.dump inside the train function, we won't have access to the model object, and probably have to define exactly what we need to save, maybe define a custom dictionary, etc. I can think how best to do overcome this.
However, what exactly are we trying to avoid here? Fundamentally, if someone does x = model.fit(..), x will be none.
But as it currently stands, if someone tries x = model.fit(... save_model = True ...), two things will happen: model will be saved in the desired folder, and x will become actually a tuple of (None, None).  One quirk is that if one calls model.fit(... save_model = False...) the return tuple will have (None, date and time). However, I can substitute datetime with a conditional None to get it to (None, None). It was just a placeholder anyway.

Do you believe it is still an issue and I should look for the workaround?

![image](https://user-images.githubusercontent.com/75586459/176228086-49b389f4-50b1-43e3-b9f5-b15dd4e8e08d.png)

Reply:
I see, as I understand it the issue is that since we **return** the result of `train_*`, any code at the end of `.fit()` outside of the conditional won't actually execute.

I know it currently "works", but I am hesitant to leave it because we should avoid ``leaking" data, and this could cause issues later on if we were to extend the methods and forget about this quirk.

Since we have access to self within the entire fit method, could we alter the code by removing the return keywords from the train_* calls, then have the save_model after, e.g.:

```
# Train inplace
if self.mode == "basic":
   train_basic(...)
elif self.mode == "wgan":
   train_wgan(...)
elif self.mode == "cgan":
   train_cgan(...)

# Save model after training
if save_model:
   pickle.dump(self,...)

# Move return to very end
return None
```
If this isn't possible, we'll keep your existing method, but if we can fix this it would be very helpful.

Reply:
Thanks Artem!

Reply:
All good -- it all gets merged back here because it's on the same branch I think! 

Will merge this evening -- thanks for this Artem!